package-tests.cake

```diff
@@ -0,0 +1,312 @@
+    protected override string PackageInstallDirectory => _packageDir + "test/nuget-netfx/";
+    protected override string PackageBinDir => PackageInstallDirectory + "tools/";
+}
+
+public class NuGetNetCorePackageTester : NetCorePackageTester
+{
+    public NuGetNetCorePackageTester(ICakeContext context, string packageVersion)
+        : base(context, packageVersion) { }
+
+    protected override string PackageName => $"NUnit.ConsoleRunner.NetCore.{_packageVersion}.nupkg";
+    protected override string PackageInstallDirectory => _packageDir + "test/nuget-netcore/";
+    protected override string PackageBinDir => PackageInstallDirectory + "tools/netcoreapp3.1/any/";
+}
+
+public class ChocolateyPackageTester : NetFXPackageTester
+{
+    public ChocolateyPackageTester(ICakeContext context, string packageVersion)
+        : base(context, packageVersion) { }
+
+    protected override string PackageName => $"nunit-console-runner.{_packageVersion}.nupkg";
+    protected override string PackageInstallDirectory => _packageDir + "test/choco/";
+    protected override string PackageBinDir => PackageInstallDirectory + "tools/";
+}
+
+public class MsiPackageTester : NetFXPackageTester
+{
+    public MsiPackageTester(ICakeContext context, string packageVersion)
+        : base(context, packageVersion)
+    {
+        // Add tests specific to the msi package
+        PackageTests.Add(new PackageTest(
+            "Run project with both copies of mock-assembly",
+            $"../../NetFXTests.nunit --config={_config}",
+            new ExpectedResult("Failed")
+            {
+                Total = 2 * 37,
+                Passed = 2 * 23,
+                Failed = 2 * 5,
+                Warnings = 0,
+                Inconclusive = 2 * 1,
+                Skipped = 2 * 7
+            }));
+    }
+
+    protected override string PackageName => $"NUnit.Console-{_packageVersion}.msi";
+    protected override string PackageInstallDirectory => _packageDir + "test/msi/";
+    protected override string PackageBinDir => PackageInstallDirectory + "NUnit.org/nunit-console/";
+
+    protected override void CreatePackageInstallDirectory()
+    {
+        // Msiexec does not tolerate forward slashes!
+        string package = PackageUnderTest.ToString().Replace("/", "\\");
+        string testDir = PackageInstallDirectory.Replace("/", "\\");
+        int rc = _context.StartProcess("msiexec", $"/a {package} TARGETDIR={testDir} /q");
+        if (rc != 0)
+            Console.WriteLine($"  ERROR: Installer returned {rc.ToString()}");
+        else
+        {
+            // Administrative install doesn't copy these files to
+            // their final destination, so we do it.
+            _context.CopyFiles(
+                PackageBinDir + "*.dll",
+                PackageBinDir + "agents/net20/");
+            _context.CopyFiles(
+                PackageBinDir + "*.dll",
+                PackageBinDir + "agents/net40/");
+        }
+    }
```

Comment:
I like this method! As a future enhancement, it might be nice to do a `nuget install` and `choco install` of those packages in a similar way, and check the final structure.

Reply:
Cool, sounds good!